client/rest-high-level/src/main/java/org/elasticsearch/client/indices/GetDataStreamResponse.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.indices;
+
+import org.elasticsearch.cluster.metadata.DataStream;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+
+
+public class GetDataStreamResponse {
+
+
+    @Override
+    public String toString() {
+        List<DataStream> thisList = new ArrayList<>(this.dataStreams);
+        thisList.sort(Comparator.comparing(DataStream::getName));
+        return "GetDataStreamResponse [dataStreams=" + thisList + "]";
+    }
+
+    private final List<DataStream> dataStreams;
+
+    GetDataStreamResponse() {
+        dataStreams = new ArrayList<>();
+    }
+
+    GetDataStreamResponse(List<DataStream> dataStreams) {
+        this.dataStreams = dataStreams;
+    }
+
+    public List<DataStream> getDataStreams() {
+        return dataStreams;
+    }
+
+
+    public static GetDataStreamResponse fromXContent(XContentParser parser) throws IOException {
+        final List<DataStream> templates = new ArrayList<>();
+        for (XContentParser.Token token = parser.nextToken(); token != XContentParser.Token.END_ARRAY; token = parser.nextToken()) {
+            if (token == XContentParser.Token.START_OBJECT) {
+                templates.add(DataStream.fromXContent(parser));
+            }
+        }
+        return new GetDataStreamResponse(templates);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(new HashSet<>(this.dataStreams));
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
```

Comment:
We usually try to avoid single statement `if` statements if I recall correctly?

Reply:
Fixed